### PR TITLE
[IOTDB-6220] Pipe: Changed the sink loopback detection logic to support hostName and IPv6 specification.

### PR DESCRIPTION
**Pull request by Caideyipi**
## [IOTDB-6220](https://issues.apache.org/jira/browse/IOTDB-6220)
Added hostName and IPv6 addresses detection of IoTDB pipe sink, and seed configNode pointing to itself.

Now the following IPs or host names pointing to the receiver on sender itself is not allowed:
1. Loopback addresses or hosts
    - "localhost"
    - "LOCALHOST"
    - "127.0.0.1"
    - "0.0.0.0.0.0.0.1"
2. General address
    - "0.0.0.0"
3. All the sender's IPv4 and IPv6 addresses
    - "192.168.100.56"
    - "10.255.1.70"
    - "fd37:634f:c2bf:421a:a63a:22aa:a80e:f5b6"
5. The senders host name
    - "DESKTOP-LOUNTUK"

### Test images
img_v3_024p_8e6311f4-f747-44fb-a86e-b8e98b05dfag
img_v3_024p_ecf8312c-f32b-49dd-9ae6-b81543d5f8cg

<hr>

This PR has:
- [ ] been self-reviewed.
    - [ ] concurrent read
    - [ ] concurrent write
    - [ ] concurrent read and write 
- [ ] added documentation for new or modified features or behaviors.
- [ ] added Javadocs for most classes and all non-trivial methods. 
- [ ] added or updated version, __license__, or notice information
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious 
  for an unfamiliar reader.
- [ ] added unit tests or modified existing tests to cover new code paths, ensuring the threshold 
  for code coverage.
- [ ] added integration tests.
- [ ] been tested in a test IoTDB cluster.

<hr>

### CODE REVIEW

File: iotdb-core/node-commons/src/main/java/org/apache/iotdb/commons/utils/NodeUrlUtils.java [hunk 1]
```diff
@@ -188,10 +193,62 @@ public static List<TConfigNodeLocation> parseTConfigNodeUrls(String configNodeUr
...
+        Arrays.stream(InetAddress.getAllByName(InetAddress.getLocalHost().getHostName()))
+            .map(InetAddress::getHostAddress)
+            .collect(Collectors.toCollection(HashSet::new));
+    // Check IPv4 and IPv6 loopback addresses 127.0.0.1 and 0.0.0.0.0.0.0.1
+    selfAddresses.addAll(
+        Arrays.stream(InetAddress.getAllByName("localhost"))
+            .map(InetAddress::getHostAddress)
+            .collect(Collectors.toCollection(HashSet::new)));
+    // Check general address 0.0.0.0
+    selfAddresses.add("0.0.0.0");
```
Comment (CRZbulabula) [inline iotdb-core/node-commons/src/main/java/org/apache/iotdb/commons/utils/NodeUrlUtils.java:250]: Don't use magic string, using private static constant instead.

File: iotdb-core/node-commons/src/main/java/org/apache/iotdb/commons/utils/NodeUrlUtils.java [hunk 2]
```diff
@@ -188,10 +193,62 @@ public static List<TConfigNodeLocation> parseTConfigNodeUrls(String configNodeUr
...
+   */
+  public static Set<String> getAllLocalAddresses() throws UnknownHostException {
+    // Check internal and external, IPv4 and IPv6 network addresses
+    Set<String> selfAddresses =
+        Arrays.stream(InetAddress.getAllByName(InetAddress.getLocalHost().getHostName()))
+            .map(InetAddress::getHostAddress)
+            .collect(Collectors.toCollection(HashSet::new));
+    // Check IPv4 and IPv6 loopback addresses 127.0.0.1 and 0.0.0.0.0.0.0.1
+    selfAddresses.addAll(
+        Arrays.stream(InetAddress.getAllByName("localhost"))
```
Comment (CRZbulabula) [inline iotdb-core/node-commons/src/main/java/org/apache/iotdb/commons/utils/NodeUrlUtils.java:246]: Don't use magic string, using private static constant instead.

**Review by CRZbulabula — COMMENTED** [review]
PTAL

**Review by CRZbulabula — APPROVED** [review]
LGTM

**Review by SteveYurongSu — APPROVED** [review]
LGTM

**Pull request merged**